caphandler/addresspools.go

```diff
@@ -34,8 +39,7 @@ func GetAddressPoolCollection(ctx iris.Context) {
 	fabricID := ctx.Params().Get("id")
 	// get all switches which are store under that fabric
 
-	_, ok := capdata.FabricDataStore.Data[fabricID]
-	if !ok {
+	if _, err := capmodel.GetFabric(fabricID); errors.Is(err, db.ErrorKeyNotFound) {
```

Comment:
We need to check for other errors also here. eg: connectivity errors, etc. Right now we are only checking for Key Not Found. Please correct in other places also.

Reply:
Incorporated comments, please review again.